doc/command_acknowledgement_codes.rst

```diff
@@ -0,0 +1,42 @@
+.. _lsst.ts.sal-acknowledgments:
+
+############################
+Command Acknowledgment Codes
+############################
+
+When programming with SAL it is important to use proper command acknowledgments codes. With consistent command acknowledgements codes project wide, we can reduce confusion and need for looking up acknowldgment definiton. *Client* in this document refers to the SAL component issuing the command, or the SAL Commander, or the salobj Remote. *Server* in this document refers to the SAL component being commanded, or the SAL Processor, or the salobj Controller.
+
+There are two ways for the Client to ask the Server for a command acknowledgement using SAL. One is waitForCommandCompletion that accepts a timeout. The other is getResponse which will immediately ask for an acknowledgement and return CMD_NOACK if none is available.
+
+Below are the list of acknowledgements you can use. In the context of DDS these values are the "ack" field of the "ackcmd" topic. To specifcy these in *ts_sal* use the code as the argument in the "ackCommand_x(ack)" function, where "x" is the command name. In ts_salobj ackcmd samples are usually created automatically, but one can create and return an ackcmd sample from a ControllerCommand callback function to override the automatically created sample.
+
+**CMD_ACK = 300**
+   Sent by the Server. The Server has accepted the command.
+
+**CMD_INPROGRESS = 301**
+   Sent by the Server. The Server has accepted the command and is processing the command.
+
+**CMD_STALLED = 302**
+   Sent by the Server. The Server has accepted the command and is expected to process for a *long* time. Useful when an application needs to cool down or warm up before it can continue. Can also send the average time expected to execute the command.
```

Comment:
Regarding "Can also send the average time expected..."

If you are going to say this then please explain how it should be done. The ackcmd topic only has 3 user-settable fields: ack (this code), error (an integer) and reason (a string). One could encode the time in the reason string or set the error to the nearest integer value, but neither is really an obvious thing to do.

@dmills-lsst is there a recommended way to return this time? If not, I think we should consider removing this sentence.

Reply:
Waiting for @dmills-lsst  response